File: frontend/src/app/core/services/brand.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Brand, BrandRequest } from '../models/brand.model';
import { MessageResponse } from '../models/common.model';

@Injectable({ providedIn: 'root' })
export class BrandService {
  private readonly http = inject(HttpClient);
  private readonly base = `${environment.apiBaseUrl}/brands`;

  getAllActive(): Observable<Brand[]> {
    return this.http.get<Brand[]>(this.base);
  }

  getFeatured(): Observable<Brand[]> {
    return this.http.get<Brand[]>(`${this.base}/featured`);
  }

  getAllForAdmin(): Observable<Brand[]> {
    return this.http.get<Brand[]>(`${this.base}/admin`);
  }

  getBySlug(slug: string): Observable<Brand> {
    return this.http.get<Brand>(`${this.base}/${slug}`);
  }

  create(request: BrandRequest): Observable<Brand> {
    return this.http.post<Brand>(this.base, request);
  }

  update(id: number, request: BrandRequest): Observable<Brand> {
    return this.http.put<Brand>(`${this.base}/${id}`, request);
  }

  delete(id: number): Observable<MessageResponse> {
    return this.http.delete<MessageResponse>(`${this.base}/${id}`);
  }

  updateStatus(id: number, active: boolean): Observable<Brand> {
    return this.http.patch<Brand>(`${this.base}/${id}/status`, { active });
  }

  uploadLogo(id: number, file: File): Observable<Brand> {
    const formData = new FormData();
    formData.append('file', file);
    return this.http.post<Brand>(`${this.base}/${id}/logo`, formData);
  }
}

File: backend/src/main/java/com/vitalora/api/service/BrandService.java
package com.vitalora.api.service;

import com.vitalora.api.dto.brand.BrandRequest;
import com.vitalora.api.dto.brand.BrandResponse;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface BrandService {
    List<BrandResponse> getAllActive();

    List<BrandResponse> getFeatured();

    List<BrandResponse> getAllForAdmin();

    BrandResponse getBySlug(String slug);

    BrandResponse create(BrandRequest request);

    BrandResponse update(Long id, BrandRequest request);

    void delete(Long id);

    BrandResponse updateStatus(Long id, boolean active);

    BrandResponse uploadLogo(Long id, MultipartFile file);
}
